pkg/kubectl/cmd/set/set_env.go

```diff
@@ -195,18 +259,39 @@ func (o *EnvOptions) Complete(f cmdutil.Factory, cmd *cobra.Command, args []stri
 	o.Output = cmdutil.GetFlagString(cmd, "output")
 	o.From = cmdutil.GetFlagString(cmd, "from")
 	o.Prefix = cmdutil.GetFlagString(cmd, "prefix")
+	o.Keys = cmdutil.GetFlagStringSlice(cmd, "keys")
 	o.DryRun = cmdutil.GetDryRunFlag(cmd)
 
-	o.EnvArgs = envArgs
-	o.Resources = resources
 	o.Cmd = cmd
 
 	o.ShortOutput = cmdutil.GetFlagString(cmd, "output") == "name"
 
+	err := o.setFromResources(f)
```

Comment:
Why do you need a separate method if this is the only place where you're using it?

Reply:
I suppose we don't really but I did need to have this callable from complete to be able to inject a specific client for testing purposes. Do you want me to move this code into the complete function? I would have thought that it might warrant its own function but happy to move it?

Reply:
In that case just:
```go
return o.setFromResources(f)
```

Reply:
Additionally, please don't pass the entire factory here, extract what you need, add it to `EnvOptions` structure and just invoke `o.setFromResources()`.